### PR TITLE
Honor test spec deployment target during validation.

### PR DESCRIPTION
**Pull request by dnkoutso**
If a test spec has a different deployment target than the library spec during validation this was not honored by the validator.

### CODE REVIEW

File: lib/cocoapods/validator.rb [hunk 1]
```diff
@@ -757,7 +759,7 @@ def test_pod
               UI.warn "Skipping test spec `#{test_spec.name}` on platform `#{consumer.platform_name}` since it is not supported.\n".yellow
             else
               scheme = @installer.target_installation_results.first[pod_target.name].native_target_for_spec(test_spec)
-              output = xcodebuild('test', scheme, 'Debug')
+              output = xcodebuild('test', scheme, 'Debug', test_spec.deployment_target(consumer.platform_name))
```
Comment (dnkoutso) [inline lib/cocoapods/validator.rb:762]: this method will also use the parent spec if none is explicitly specified.

File: lib/cocoapods/validator.rb [hunk 2]
```diff
@@ -1048,7 +1050,7 @@ def parse_xcodebuild_output(output)
     # @return [String] Executes xcodebuild in the current working directory and
     #         returns its output (both STDOUT and STDERR).
     #
-    def xcodebuild(action, scheme, configuration)
+    def xcodebuild(action, scheme, configuration, deployment_target)
```
Comment (amorde) [inline lib/cocoapods/validator.rb:1053]: lol love how this is the only change to the method. variable shadowing ftw?

**Comment by paulb777** [comment]
Does this fix #9524?

File: lib/cocoapods/validator.rb (hunk 2 above)
Comment (segiddins) [reply lib/cocoapods/validator.rb:1053]: nit: use a named parameter ?

File: spec/unit/validator_spec.rb [hunk 3]
```diff
@@ -600,6 +600,39 @@ def podspec_path(name = 'JSONKit', version = '1.4')
         validator.results_message.strip.should.be.empty
       end
 
+      it 'uses the deployment target of the current test spec' do
```
Comment (segiddins) [inline spec/unit/validator_spec.rb:603]: it's not super clear to me what in here is validating that the correct deployment target is being used for xcodebuild ?

File: lib/cocoapods/validator.rb (hunk 2 above)
Comment (dnkoutso) [reply lib/cocoapods/validator.rb:1053]: love it

File: spec/unit/validator_spec.rb (hunk 3 above)
Comment (amorde) [reply spec/unit/validator_spec.rb:603]: it pulls a destination from `Fourflusher` once requesting iOS 8 for building the library and once requesting iOS 9 to run the test. at least that was my understanding. maybe would be better to add an expectation that `xcodebuild` is provided the correct deployment targets instead

Comment (dnkoutso) [reply spec/unit/validator_spec.rb:603]: yeap. Those tests are super convoluted and hard to follow. I verified without my change this test fails.
